cibuildwheel/projectfiles.py

```diff
@@ -106,3 +104,20 @@ def get_requires_python_str(package_dir: Path) -> str | None:
         return setup_py_python_requires(f2.read())
 
     return None
+
+
+def resolve_dependency_groups(pyproject_toml: dict[str, Any], *groups: str) -> tuple[str, ...]:
+    """
+    Get the packages in dependency-groups for a package.
+    """
+
+    if not groups:
+        return ()
+
+    try:
+        dependency_groups_toml = pyproject_toml["dependency-groups"]
+    except KeyError:
+        msg = f"Didn't find [dependency-groups], which are needed to resolve {groups!r}."
```

Comment:
Should `pyproject.toml` be mentioned here ?
Should its existence be tested (we could allow  `dict[str, Any] | None` as an argument here) to provide another error message in case it's completely missing ?